src/routes/api/issues.ts

```diff
@@ -1,75 +1,66 @@
-    const { timestamp, message, roomId, sectionId } = request.body;
-    const room = await roomModel.findById(roomId); //parsing for room id
-    const section = await sectionModel.find({ sections: sectionId }); //parsing for room id
-    const ts = Date.parse(timestamp); //parsing timestamp to store and convert / check if proper date type
-    const user = await userModel.findOne({ email: request.userEmail });
+const issueRoute = async (req: Request, res: Response): Promise<void> => {
+    const { message, bookingId } = req.body;
 
-    //if the room is valid
-    if (section === null) {
-        //if the room's section isn't valid / isn't found in database
-        response.status(422).json({ error: "That section doesn't exist, or was inputted incorrectly." });
+    const timeBlock = await TimeBlockModel.findOne({ _id: bookingId });
+    if (!timeBlock) {
+        res.status(400);
         return;
     }
-    //if the section is valid
-    if (typeof message !== 'string') {
-        //checks to ensure that the message is a string type to prevent attacks
-        response.status(422).json({ error: "That message isn't the proper format!" });
+    const section = await SectionModel.findOne({ _id: timeBlock.sectionId });
+    if (!section) {
+        res.status(400);
         return;
     }
-    //if the input type for message is string
-    if (ts === null) {
-        response.status(422).json({ error: "That timestamp isn't the proper format!" });
-        return;
-    }
-    if (user === null) {
-        response.status(422).json({ error: 'User not found!' });
+    const room = await RoomModel.findOne({ _id: section.roomId });
+    if (!room) {
+        res.status(400);
         return;
     }
 
-    const issue = new issueModel({
-        timestamp: ts,
+    const issue = await IssueModel.create({
+        reportingUserEmail: req.userEmail,
         message: message,
-        status: 'Unresolved',
-        roomId: roomId,
-        sectionId: sectionId,
-        reportingUserId: user.id,
+        status: 'unresolved',
+        bookingId: bookingId,
     });
-    if (room === null) {
-        //if the room isn't valid / isn't found in database
-        response.status(422).json({ error: "That room doesn't exist, or was inputted incorrectly." });
-        return;
-    } else {
-        //if room exists, push the issue id as foreign key for future population in querying
-        room.issues.push(issue.id);
+
+    room.issues.push(issue._id);
+    await room.save();
+
+    const possibleTimeBlocks = await TimeBlockModel.find({ sectionId: timeBlock.sectionId, startsAt: { $lte: timeBlock.startsAt } });
+    let maxDate = new Date(0);
+    let blamedUsers;
+    for (const possibleTimeBlock of possibleTimeBlocks) {
+        if (possibleTimeBlock.startsAt > maxDate) {
+            maxDate = possibleTimeBlock.startsAt;
+            blamedUsers = possibleTimeBlock.users;
+        }
```

Comment:
Is the intended behavior of this to set blamedUsers to the last value of possibleTimeBlocks? If so, we can get rid of the loop and just take the last index (or even better, update our query to take the most recent).